### PR TITLE
fix(formfield): add brain form-field and forms into the generator CLI

### LINKED ISSUE #358: Public alias of form-field-helm seems wrong
### Please provide the environment you discovered this bug in.

Angular 18, Nx 19

### Which area/package is the issue in?

Don't know / other

### Description

[form-field](https://github.com/goetzrobin/spartan/blob/30bf1fa0159b9534823db66cdc65a756b38089cc/libs/cli/src/generators/ui/libs/ui-form-field-helm/generator.ts#L10)
`publicName` is `ui-formfield-helm` (missing dash). 
Executing `npx nx g @spartan-ng/cli:ui form-field` generates a `tsconfig` path alias of `@spartan-ng/ui-formfield-helm`, which contradicts the documentation of the `form-field` package [here](https://www.spartan.ng/components/form-field) resulting in a compilation error of a wrong import.
```ts

import { HlmFormFieldModule } from '@spartan-ng/ui-form-field-helm';
import { HlmInputDirective } from '@spartan-ng/ui-input-helm';

```

### Please provide the exception or error you saw

```true
✘ [ERROR] TS2307: Cannot find module '@spartan-ng/ui-form-field-helm' or its corresponding type declarations. [plugin angular-compiler]

    libs/product-card/src/lib/product-card/product-card.component.ts:4:35:
      4 │ import { HlmFormFieldModule } from '@spartan-ng/ui-form-field-helm';
        ╵                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Other information

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by cjosue15**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [x] form-field
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Currently when we added the formfield component or the input through the CLI, the ui-form-field-brain and ui-forms-brain packages were not being added as peerDependencies. 

Closes #358, #351, #365

## What is the new behavior?

The CLI will now automatically add the packages when the formfield or input is installed.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: libs/cli/src/generators/ui/libs/ui-form-field-helm/files/lib/form-field.spec.ts.template
Comment (goetzrobin) [inline libs/cli/src/generators/ui/libs/ui-form-field-helm/files/lib/form-field.spec.ts.template:1]: what's the reason for deleting those?

File: libs/cli/src/generators/ui/supported-ui-libraries.json [hunk 1]
```diff
@@ -332,7 +334,8 @@
 			"@angular/core": "^18.0.0",
 			"@spartan-ng/ui-core": "0.0.1-alpha.353",
 			"class-variance-authority": "^0.6.0",
-			"clsx": "^1.2.1"
+			"clsx": "^1.2.1",
+			"@spartan-ng/ui-form-field-brain": "0.0.1-alpha.353"
```
Comment (goetzrobin) [inline libs/cli/src/generators/ui/supported-ui-libraries.json:338]: I think the select component should get this also?

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Couple questions! Thanks so much for your work!

File: libs/cli/src/generators/ui/libs/ui-form-field-helm/files/lib/form-field.spec.ts.template
Comment (cjosue15) [reply libs/cli/src/generators/ui/libs/ui-form-field-helm/files/lib/form-field.spec.ts.template:1]: I thought we did not include the tests, but reviewing the whole project I saw some tests included in the components. So i will add it again.

File: libs/cli/src/generators/ui/supported-ui-libraries.json (hunk 1 above)
Comment (cjosue15) [reply libs/cli/src/generators/ui/supported-ui-libraries.json:338]: Sure, good catch!

File: libs/cli/src/generators/ui/libs/ui-form-field-helm/files/lib/form-field.spec.ts.template
Comment (goetzrobin) [reply libs/cli/src/generators/ui/libs/ui-form-field-helm/files/lib/form-field.spec.ts.template:1]: sorry I might have confused something here. I think removing the test is okay. I was wondering why the other templates where deleted for the hlm part of the formfield?

**Pull request merged**